src/server/auth/server/api_server.go

```diff
@@ -1707,6 +1720,101 @@ func (a *apiServer) GetScope(ctx context.Context, req *auth.GetScopeRequest) (re
 	return response, nil
 }
 
+func (a *apiServer) SetDefaultACL(ctx context.Context, req *auth.SetDefaultACLRequest) (*auth.SetDefaultACLResponse, error) {
+	a.LogReq(req)
+
+	if a.activationState() == none {
+		return nil, auth.ErrNotActivated
+	}
+
+	// Only cluster admins can set the default ACL
+	callerInfo, err := a.getAuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	callerIsAdmin, err := a.hasClusterRole(ctx, callerInfo.Subject, auth.ClusterRole_SUPER)
+	if err != nil {
+		return nil, err
+	}
+
+	if !callerIsAdmin {
+		return nil, &auth.ErrNotAuthorized{
+			Subject: callerInfo.Subject,
+			AdminOp: "SetDefaultACL for cluster",
+		}
+	}
+
+	principal, err := a.canonicalizeSubject(ctx, req.Username)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := col.NewSTM(ctx, a.env.GetEtcdClient(), func(stm col.STM) error {
+		acl := a.defaultAcl.ReadWrite(stm)
+		var existing auth.ACL
+		err := acl.Get(defaultAclKey, &existing)
+		if err != nil {
+			if !col.IsErrNotFound(err) {
+				return err
+			}
+		}
+		if existing.Entries == nil {
+			existing.Entries = make(map[string]auth.Scope)
+		}
+		if req.Scope == auth.Scope_NONE {
+			delete(existing.Entries, principal)
+		} else {
+			existing.Entries[principal] = req.Scope
```

Comment:
I know I always nitpick over names (sorry!) but FWIW the semantics here look more like the `SetScope` API than the `SetACL` API. Do you think it would be worth naming it something like `SetDefaultScope`?